Modules/MUON/MID/src/MIDTrending.cxx

```diff
@@ -0,0 +1,177 @@
+
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file MIDTrending.cxx
+/// \author Valerie Ramillien     based on Piotr Konopka
+///
+
+#include "MID/MIDTrending.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/DatabaseInterface.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Reductor.h"
+#include "QualityControl/ObjectMetadataKeys.h"
+#include "QualityControl/RootClassFactory.h"
+#include "QualityControl/RepoPathUtils.h"
+#include <TDatime.h>
+#include <TH1.h>
+#include <TCanvas.h>
+#include <TPaveText.h>
+#include <TGraphErrors.h>
+#include <TPoint.h>
+#include <map>
+#include <string>
+#include <memory>
+#include <vector>
+using namespace o2::quality_control;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::repository;
+using namespace o2::quality_control::postprocessing;
+
+void MIDTrending::configure(const boost::property_tree::ptree& config)
+{
+  mConfig = TrendingTaskConfigMID(getID(), config);
+}
+
+void MIDTrending::initialize(Trigger, framework::ServiceRegistryRef services)
+{
+  // Preparing data structure of TTree
+  mTrend = std::make_unique<TTree>();
+  mTrend->SetName(PostProcessingInterface::getName().c_str());
+  mTrend->Branch("runNumber", &mMetaData.runNumber);
+
+  mTrend->Branch("ntreeentries", &ntreeentries);
+  mTrend->Branch("time", &mTime);
+
+  for (const auto& source : mConfig.dataSources) {
+    std::unique_ptr<Reductor> reductor(root_class_factory::create<Reductor>(source.moduleName, source.reductorName));
+
+    mTrend->Branch(source.name.c_str(), reductor->getBranchAddress(), reductor->getBranchLeafList());
+    mReductors[source.name] = std::move(reductor);
+  }
+}
+
+// todo: see if OptimizeBaskets() indeed helps after some time
+void MIDTrending::update(Trigger t, framework::ServiceRegistryRef services)
+{
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+  trendValues(t, qcdb);
+  generatePlots(qcdb);
+}
+
+void MIDTrending::finalize(Trigger, framework::ServiceRegistryRef services)
+{
+
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+  auto mo = std::make_shared<core::MonitorObject>(mTrend.get(), getName(), "o2::quality_control_modules::trd::MIDTrending", mConfig.detectorName);
```

Comment:
```suggestion
  auto mo = std::make_shared<core::MonitorObject>(mTrend.get(), getName(), "o2::quality_control_modules::mid::MIDTrending", mConfig.detectorName);
```
Typo?

Reply:
Yes used TRDTrending as a template for MID and forgot this modification